### PR TITLE
Fixed #59368

### LINKED ISSUE #59368: Explorer should allow basename with dot
VSCode Version: v1.26.1 (1.26.1)
OS Version: macOS High Sierra v0.13.6
When doing @reasonml, it's very annoying to have all those **.bs.js files generated (just like it was with @typescriptlang at the beginning), so I tried to ignore them in my file tree but it doesn't seem to work.

Steps to Reproduce:

1. Add this to your settings.json
```
{
  "files.exclude": {
    "**/*.bs.js": { "when": "$(basename).re" }
  }
}
```
2. Reload, go to file explorer and see that you can still see the *.bs.js generated files. ⚠️

Does this issue occur when all extensions are disabled?: **Yes**

### LINKED ISSUE #59378: No regex validator for terminal find widget
#59267

We can add the same regex validator for the terminal find widget input box if the underling regex engine is JavaScript regex. Close this one if we are using another regex engine somehow in xterm.js.

### PR DESCRIPTION
**Pull request by irrationalRock**
This PR fixes: https://github.com/Microsoft/vscode/issues/59378.

It adds regex validation to the terminal.

### CODE REVIEW

File: src/vs/editor/contrib/find/simpleFindWidget.ts [hunk 1]
```diff
@@ -44,6 +45,22 @@ export abstract class SimpleFindWidget extends Widget {
 		this._findInput = this._register(new ContextScopedFindInput(null, this._contextViewService, {
 			label: NLS_FIND_INPUT_LABEL,
 			placeholder: NLS_FIND_INPUT_PLACEHOLDER,
+			validation: (value: string): InputBoxMessage | null => {
+				if (value.length === 0) {
+					return null;
+				}
+				if (!this._findInput.getRegex()) {
```
Comment (rhogeranacleto) [inline src/vs/editor/contrib/find/simpleFindWidget.ts]: What do you think about to put both of those conditions in one _if_ with an _OR_ statement?

File: src/vs/editor/contrib/find/simpleFindWidget.ts [hunk 2]
```diff
@@ -44,6 +45,22 @@ export abstract class SimpleFindWidget extends Widget {
...
+					return null;
+				}
+				try {
+					/* tslint:disable:no-unused-expression */
+					new RegExp(value);
```
Comment (rhogeranacleto) [inline src/vs/editor/contrib/find/simpleFindWidget.ts]: Could be more simple use `tslint:disable-next-line:no-unused-expression` instead disable and enable next :slightly_smiling_face:

**Comment by irrationalRock** [comment]
Thanks for the suggestions, 

I also applied these changes to `FindWidget`.

**Comment by alexandrudima** [comment]
Hey Joao, @rebornix owns the find widget since around 2 years.

**Review by rebornix — APPROVED** [review]
@irrationalRock thanks for your contribution!

**Pull request merged**